### PR TITLE
fine tuning in kava and phoenix analytics creation update

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: tvplayer/src/main/java/com/kaltura/tvplayer/PlayerTokenResolver.java [hunk 1]
```diff
@@ -12,19 +14,33 @@
...
+            if (mediaEntry.getMediaType() != null) {
+                set("entryType", mediaEntry.getMediaType().name());
+            }
+
+            if (ks != null) {
+                set("ks", ks);
+            }
+
+            if (TextUtils.isDigitsOnly(mediaEntry.getId())) /* OTT Media */ {
+                set("entryId", isVeaidEntryIdInMetadata(mediaEntryMetadata) ? mediaEntryMetadata.get("entryId") : "unknown");
```
Comment (noamtamim) [inline tvplayer/src/main/java/com/kaltura/tvplayer/PlayerTokenResolver.java]: isVeaidEntryIdInMetadata => typo

File: tvplayer/src/main/java/com/kaltura/tvplayer/PlayerTokenResolver.java [hunk 2]
```diff
@@ -38,22 +54,31 @@ void update(PKMediaEntry mediaEntry) {
         }
     }
 
+    private boolean isVeaidEntryIdInMetadata(Map<String, String> mediaEntryMetadata) {
+        return mediaEntryMetadata != null && mediaEntryMetadata.containsKey("entryId") && !TextUtils.isEmpty(mediaEntryMetadata.get("entryId"));
```
Comment (noamtamim) [inline tvplayer/src/main/java/com/kaltura/tvplayer/PlayerTokenResolver.java]: no need for contains() -- get(key) will return null.

File: tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java [hunk 3]
```diff
@@ -117,18 +115,22 @@ protected KalturaPlayer(Context context, PlayerInitOptions initOptions) {
             this.preload = true; // autoplay implies preload
         }
         this.referrer = buildReferrer(context, initOptions.referrer);
-        if (kalturaPlayerType == KalturaPlayerType.basic || (kalturaPlayerType == KalturaPlayerType.ott && initOptions.uiConfPartnerId == null)) {
+        if (kalturaPlayerType == KalturaPlayerType.basic || (kalturaPlayerType == KalturaPlayerType.ott && (initOptions.phoenixTVPlayerDMSParams == null || (initOptions.phoenixTVPlayerDMSParams != null && initOptions.phoenixTVPlayerDMSParams.ovpPartnerId == null)))) {
```
Comment (noamtamim) [inline tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java]: Extract method? "if" is too long.

File: tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java [hunk 4]
```diff
@@ -569,6 +575,8 @@ public void loadMedia(OVPMediaOptions mediaOptions, final OnEntryLoadListener li
 
         if (mediaOptions.ks != null) {
```
Comment (noamtamim) [inline tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java]: Extract method updateKS() with the below

File: tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java [hunk 5]
```diff
@@ -593,6 +605,8 @@ public void loadMedia(OTTMediaOptions mediaOptions, final OnEntryLoadListener li
...
```
Comment (noamtamim) [inline tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java]: Extract method updateKS() with the above

File: tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java [hunk 6]
```diff
@@ -642,70 +656,47 @@ private boolean isOTTPlayer() {
...
-
     private PhoenixAnalyticsConfig getPhoenixAnalyticsConfig() {
         // Special case: Phoenix plugin
         String name = PhoenixAnalyticsPlugin.factory.getName();
-        JsonObject phoenixAnalyticObject;
-        if (getInitOptions().pluginConfigs.hasConfig(name)) {
-            phoenixAnalyticObject = (JsonObject) getInitOptions().pluginConfigs.getPluginConfig(name);
-        } else {
-            phoenixAnalyticObject = phoenixAnalyticDefaults(getPartnerId(), getServerUrl(), getKS(), Consts.DEFAULT_ANALYTICS_TIMER_INTERVAL_HIGH_SEC);
+        PKPluginConfigs pkPluginConfigs = getInitOptions().pluginConfigs;
```
Comment (noamtamim) [inline tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java]: Don't allow config in json.

**Review by noamtamim — CHANGES_REQUESTED** [review]

File: tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java (hunk 6 above)
Comment (giladna) [reply tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java]: done

**Review by noamtamim — APPROVED** [review]

**Review by noamtamim — APPROVED** [review]

**Pull request merged**